core/src/main/scala/akka/persistence/jdbc/state/scaladsl/JdbcDurableStateStore.scala

```diff
@@ -0,0 +1,72 @@
+package akka.persistence.jdbc.state.scaladsl
+
+import scala.concurrent.{ ExecutionContext, Future }
+import slick.jdbc.{ JdbcBackend, JdbcProfile }
+import akka.Done
+import akka.persistence.state.scaladsl.{ DurableStateUpdateStore, GetObjectResult }
+import akka.serialization.Serialization
+import akka.persistence.jdbc.state.{ AkkaSerialization, DurableStateQueries }
+import akka.persistence.jdbc.config.DurableStateTableConfiguration
+import akka.persistence.jdbc.state.DurableStateTables
+import akka.dispatch.ExecutionContexts
+
+class JdbcDurableStateStore[A](
+    db: JdbcBackend#Database,
+    profile: JdbcProfile,
+    durableStateConfig: DurableStateTableConfiguration,
+    serialization: Serialization)(implicit ec: ExecutionContext)
+    extends DurableStateUpdateStore[A] {
+  import profile.api._
+
+  val queries = new DurableStateQueries(profile, durableStateConfig)
+
+  def getObject(persistenceId: String): Future[GetObjectResult[A]] =
+    db.run(queries._selectByPersistenceId(persistenceId).result).map { rows =>
+      rows.headOption match {
+        case Some(row) =>
+          GetObjectResult(AkkaSerialization.fromRow(serialization)(row).toOption.asInstanceOf[Option[A]], row.seqNumber)
+
+        case None =>
+          GetObjectResult(None, 0)
+      }
+    }
+
+  def upsertObject(persistenceId: String, seqNr: Long, value: A, tag: String): Future[Done] = {
+    require(seqNr > 0)
+    val row =
+      AkkaSerialization.serialize(serialization, value).map { serialized =>
+        DurableStateTables.DurableStateRow(
+          persistenceId,
+          seqNr,
+          serialized.payload,
+          if (tag.isEmpty()) None else Some(tag),
```

Comment:
Are we sure `tag` won't be null? 

Maybe 

```suggestion
          Option(tag).filter(_.trim.nonEmpty),
```

Reply:
done